docs/en/ingest-management/fleet/air-gapped.asciidoc

```diff
@@ -32,6 +32,43 @@ Set the following property in {kib} to enable air-gapped mode in {fleet}. This a
 xpack.fleet.isAirGapped: true
 ----
 
+[discrete]
+[[air-gapped-pgp-fleet]]
+== Configure {agents} to download a PGP/GPG key from {fleet-server}
+
+Starting from version 8.9.0, when {agent} tries to perform an upgrade, it first verifies the binary signature with the key bundled in the agent. This process has a backup mechanism that will use the key coming from `https://artifacts.elastic.co/GPG-KEY-elastic-agent` instead of the one it already has.
+
+In an air-gapped environment, the agent won't be able to download the remote key and therefore cannot be upgraded. For versions 8.9.0 to 8.10.3, you can resolve this problem following the steps described in the associated link:https://www.elastic.co/guide/en/fleet/8.9/release-notes-8.9.0.html#known-issues-8.9.0[known issue] documentation.
+
+
+Starting in version 8.10.4, you can resolve this problem by configuring {agents} to download the PGP/GPG key from {fleet-server}.
+
+By default, {fleet-server} serves {agents} with the key located in `FLEETSERVER_BINARY_DIR/elastic-agent-upgrade-keys/default.pgp`.
+The key is served through the endpoint `FLEETSERVER_BINARY_DIR/elastic-agent-upgrade-keys/default.pgp`.
+Accessing the endpoint requires a valid API key and the endpoint is rate limited.
+
+If there isn't a `default.pgp` key in the `FLEETSERVER_BINARY_DIR/elastic-agent-upgrade-keys/default.pgp` directory, {fleet-server} instead will attempt to retrieve a PGP/GPG key from the URL that you can specify with the `server.pgp.upstream_url` setting.
```

Comment:
```suggestion
If there isn't a `default.pgp` key in the `FLEETSERVER_BINARY_DIR/elastic-agent-upgrade-keys/default.pgp` directory, {fleet-server} instead will attempt to retrieve a PGP/GPG key from the URL that you can specify with the `server.pgp.upstream_url` setting.
```

Few points to clarify:

- Is the `default.pgp` bundled with Fleet Server?

If it's bundled, air-gapped users are good to go with it.

- Is the `default.pgp` downloaded lazily from `https://artifacts.elastic.co/GPG-KEY-elastic-agent` by default?

If it is the case, then air-gapped users can grant access to it (HTTP Proxy or open the firewall).

- Is the `default.php` downloaded lazily from the URL provided via `server.pgp.upstream_url` setting?

If it is the case, then air-gapped users can specify a URL so that it will be downloaded from there.

Reply:
It's a lazy download from `server.pgp.upstream_url`

Reply:
To be very clear: 
> * Is the `default.pgp` bundled with Fleet Server?

No, on install this file does not exist; but a user can add it manually write the file on the instance if there is 0 internet access in order to distribute it to the agents

> * Is the `default.pgp` downloaded lazily from `https://artifacts.elastic.co/GPG-KEY-elastic-agent` by default?

Yes, the default value of `server.pgp.upstream_url` is `https://artifacts.elastic.co/GPG-KEY-elastic-agent`; if `default.pgp` is not on-disk when the key is requested, fleet-server will download the file from the upstream URL, save it to disk and serve it

> * Is the `default.php` downloaded lazily from the URL provided via `server.pgp.upstream_url` setting?

yes, see above 😄